aiohttp/payload.py

```diff
@@ -278,6 +296,30 @@ def __init__(self, value,
             content_type=content_type, encoding=encoding, *args, **kwargs)
 
 
+class AsyncIterablePayload(Payload):
+
+    def __init__(self, value, *args, **kwargs):
+        assert isinstance(value, AsyncIterable), \
+            ("value argument must support "
+             "collections.abc.AsyncIterablebe interface, "
+             "got {!r}".format(value))
+
+        if 'content_type' not in kwargs:
+            kwargs['content_type'] = 'application/octet-stream'
+
+        super().__init__(value, *args, **kwargs)
+
+        self._iter = value.__aiter__()
+
+    async def write(self, writer):
+        try:
+            while True:
```

Comment:
Here need a check that `self._iter` is not None for the case when the same instance of this class will be used twice.

Reply:
Done